src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -112,10 +113,38 @@ ErrorIfRelationIsAKnownShard(Oid relationId)
 	}
 
 	const char *relationName = get_rel_name(relationId);
+
 	ereport(ERROR, (errmsg("relation \"%s\" is a shard relation ", relationName)));
 }
 
 
+/*
+ * ErrorIfChangingKnownShard errors out if the relation with relationId is
+ * a known shard and manual changes on known shards are disabled. This is
+ * valid for only non-citus (external) connections.
+ */
+void
+ErrorIfChangingKnownShard(Oid relationId)
```

Comment:
Minor:

When we call `ErrorIfChangingKnownShard` I would expect this method to error if the given relation belongs to a known shard but this won't happen if `EnableManualChangesToShards` is false.

So I think it makes more sense to either change the function name slightly to reflect that behavior or have  the `EnableManualChangesToShards` check when calling this function:

``` c
if (!IsManualChangeToShardsEnabled) {
     ErrorIfChangingKnownShard(relationId)
}
```

Reply:
Renamed to `ErrorIfIllegallyChangingKnownShard`